src/engine.rs

```diff
@@ -44,15 +44,19 @@ pub type EngineResult<T> = Result<T, EngineError>;
 pub trait Dev: Debug {
     fn copy(&self) -> Box<Dev>;
     fn get_id(&self) -> String;
+    fn eq(&self, other: &Path) -> bool;
```

Comment:
This is not the usual definition of eq. Something should give, either the name or the type of other.